### PR TITLE
39 streamlining serial job script

### PR DESCRIPTION
**Pull request by tbctan**
generated `atom_run_job.sh` job now looks like (where 9 is the # rows of the list_of_single_atoms.csv file)

```
# SERIAL FORMAT

# Run system with idx 0 to 9
for idx in $(seq 0 9); do
   id=$(printf "%04d" $idx)
   pushd /home2/bctan/STATE-BATCH/STATE-BATCH/examples/single_atom/atom_run_$id
   ./STATE < statebatch.in > statebatch.out
   popd
done
```

### CODE REVIEW

File: statebatch/statebatch/jobutils.py [hunk 1]
```diff
@@ -35,13 +42,20 @@ def command(exe_command, input_file, output_file):
...
+
         with open(bashfile, 'w') as fo:
             fo.write("# SERIAL FORMAT"+'\n\n')
-            for idx, val in jobinfo.items():
-                fo.write(f"# Run system with idx {idx:04d}" + '\n')
-                fo.write(f"pushd {val['cwd']}"+'\n')
-                fo.write(command(exe_command, val['input_file'], val['output_file']) + '\n')
-                fo.write("popd" + '\n\n')
+            fo.write(f"# Run system with idx 0 to {len_idx}" + '\n')
+            fo.write(f"for idx in $(seq 0 {len_idx}); do" + '\n')
```
Comment (hhlim12) [inline statebatch/statebatch/jobutils.py:53]: From last meeting, we discuss that the index in jobscript in running directory should start from 1.
So the seq index should be from 1 to {len_idx}?

Comment (tbctan) [reply statebatch/statebatch/jobutils.py]: @hhlim12 This refers to the atom index in the folders generated, which currently starts from atom..._0000. Will this also change to atom..._0001 as the starting atom? If so, then 1 to {len_idx} is correct. I committed the change then realized that the atom..._000x folders still start from 0.

Comment (hhlim12) [reply statebatch/statebatch/jobutils.py]: Sorry. Let me check again and do simple run.

Comment (kimrojas) [reply statebatch/statebatch/jobutils.py]: can this be reimplemented via python instead of bash ?
Maybe via os.system or subprocess.run module

**Comment by kimrojas** [comment]
@tbctan Can you try to refactor the implementation. 
From bash to python script ? 
Maybe copy how the parallel version is implemented
Parallel version = take index > read command from job dict with specific index > run
In new serial = NO INDEX > read all commands from job dict > run all commands in loop

**Comment by tbctan** [comment]
@kimrojas Hi, requesting review for merge. I changed implementation to python in jobutils.py. I have also made the .json dictionary processing global for both serial and parallel as opposed to implementing them individually. 

This still needs MPI implementation in serial, for which I will open a new issue.

**Review by kimrojas — APPROVED** [review]
Looks good, I recoded the python file a little. 
Can you please cleanup unused code ?
Thanks!

**Comment by tbctan** [comment]
Cleaned up unused code and related comments.

**Pull request merged**